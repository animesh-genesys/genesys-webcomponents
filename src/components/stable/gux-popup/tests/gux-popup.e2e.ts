import { newSparkE2EPage, a11yCheck } from '../../../../../tests/e2eTestUtils';

function getGuxPopupHtml(expanded: boolean, disabled: boolean): string {
  return `
  <gux-popup ${expanded ? 'expanded' : ''} ${disabled ? 'disabled' : ''}>
    <gux-button slot="target">Target</gux-button>
    <gux-list slot="popup">
      <gux-list-item>Item 1</gux-list-item>
      <gux-list-item>Item 2</gux-list-item>
      <gux-list-item>Item 3</gux-list-item>
    </gux-list>
  </gux-popup>
`;
}

describe('gux-popup', () => {
  [
    { expanded: false, disabled: false },
    { expanded: true, disabled: false },
    { expanded: false, disabled: true },
    { expanded: true, disabled: true }
  ].forEach(({ expanded, disabled }, index) => {
    it(`should display component as expected (${index + 1})`, async () => {
      const html = getGuxPopupHtml(expanded, disabled);
      const page = await newSparkE2EPage({ html });
      const element = await page.find('gux-popup');
      await a11yCheck(page);

      expect(element.outerHTML).toMatchSnapshot();
    });
  });
});
