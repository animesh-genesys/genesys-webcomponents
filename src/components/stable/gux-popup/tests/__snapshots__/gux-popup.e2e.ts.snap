// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gux-popup should display component as expected (1) 1`] = `"<gux-popup hydrated=\\"\\"> <gux-button slot=\\"target\\" hydrated=\\"\\">Target</gux-button> <gux-list slot=\\"popup\\" hydrated=\\"\\"> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 1</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 2</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 3</gux-list-item> </gux-list> </gux-popup>"`;

exports[`gux-popup should display component as expected (2) 1`] = `"<gux-popup expanded=\\"\\" hydrated=\\"\\"> <gux-button slot=\\"target\\" hydrated=\\"\\">Target</gux-button> <gux-list slot=\\"popup\\" hydrated=\\"\\"> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 1</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 2</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 3</gux-list-item> </gux-list> </gux-popup>"`;

exports[`gux-popup should display component as expected (3) 1`] = `"<gux-popup disabled=\\"\\" hydrated=\\"\\"> <gux-button slot=\\"target\\" hydrated=\\"\\">Target</gux-button> <gux-list slot=\\"popup\\" hydrated=\\"\\"> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 1</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 2</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 3</gux-list-item> </gux-list> </gux-popup>"`;

exports[`gux-popup should display component as expected (4) 1`] = `"<gux-popup expanded=\\"\\" disabled=\\"\\" hydrated=\\"\\"> <gux-button slot=\\"target\\" hydrated=\\"\\">Target</gux-button> <gux-list slot=\\"popup\\" hydrated=\\"\\"> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 1</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 2</gux-list-item> <gux-list-item role=\\"listitem\\" hydrated=\\"\\">Item 3</gux-list-item> </gux-list> </gux-popup>"`;
